### ISSUE TITLE
The `bench/generation/metrics/perplexity.py` will produce a wrong result when n_ctx > n_batch

### ISSUE THREAD

**Issue by yrom** [huggingface/optimum-quanto#391]
If n_ctx > n_batch,  e.g. `n_ctx=1024`, `n_batch=512`, then `num_batches=2`
https://github.com/huggingface/optimum-quanto/blob/157f419c2c5c72fe5dadd302255380c0644e5e78/bench/generation/metrics/perplexity.py#L157

But begin from line 186 to 193, the codes for computing probability:

https://github.com/huggingface/optimum-quanto/blob/157f419c2c5c72fe5dadd302255380c0644e5e78/bench/generation/metrics/perplexity.py#L186-L193

It will only iterate through the first element of `logits` (which is a python `list`!)

Maybe the `logits` should be `concat`-ed before the loop?
